uqcsbot/scripts/umart.py

```diff
@@ -0,0 +1,81 @@
+from uqcsbot import bot, Command
+from requests import get
+from requests.exceptions import RequestException
+from contextlib import closing
+from bs4 import BeautifulSoup
+
+NO_QUERY_MESSAGE = "You can't look for nothing. `!umart <QUERY>`"
+NO_RESULTS_MESSAGE = "I can't find anything. Try `!umart <SOMETHING NOT AS SPECIFIC>`"
+ERROR_MESSAGE = "I tried to get the things but alas I could not. Error with HTTP Request."
+
+UMART_SEARCH_URL = "https://www.umart.com.au/umart1/pro/products_list_searchnew_min.phtml"
+UMART_PRODUCT_URL = "https://www.umart.com.au/umart1/pro/"
+
+
+@bot.on_command("umart")
+def handle_umart(command: Command):
+    """
+    `!umart <QUERY>` - Returns 5 top results for products from umart matching the search query.
+    """
+    # Makes sure the query is not empty
+    if not command.has_arg():
+        bot.post_message(command.channel_id, NO_QUERY_MESSAGE)
+        return
+    search_query = command.arg.strip()
+    # Detects if user is being smart
+    if "SOMETHING NOT AS SPECIFIC" in search_query:
+        bot.post_message(command.channel_id, "Not literally...")
+        return
+    search_results = get_umart_results(search_query)
+    if not search_results:
+        bot.post_message(command.channel_id, ERROR_MESSAGE)
+        return
+    if len(search_results) == 0:
+        bot.post_message(command.channel_id, NO_RESULTS_MESSAGE)
+        return
+    message = "```"
+    for result in search_results:
+        message += f"Name: <{UMART_PRODUCT_URL}{result['link']}|{result['name']}>\n"
+        message += f"Price: {result['price']}\n"
+    message += "```"
+    bot.logger.info(message)
+    bot.post_message(command.channel_id, message)
+
+
+def get_umart_results(search_query):
+    """
+    Gets the top results based on the search_query.
+    Returns up to 5 results.
+    """
+    search_page = get_search_page(search_query)
+    if search_page is None:
+        return None
+    search_results = get_results_from_page(search_page)
+    return search_results
```

Comment:
can inline this imo since the function name is fairly descriptive of the action (i.e. `return get_results blah blah`)